### DISCUSSION TITLE
Compass shows true north

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#628] [Q&A] [answered]
We can allow the compass to show the direction of the [true north](https://en.wikipedia.org/wiki/True_north).

This is mainly useful in normal map view, as in navigation / follow modes with map rotation,
the compass already rotates to show the north direction of the map.

It will be optional in "Settings | Map | Compass sensor",
as sensors affect the battery.

---

**True north** is the direction that points directly towards the geographic North Pole. This is a fixed point on the Earth’s globe.

**Magnetic north** is the direction that a compass needle points to as it aligns with the Earth’s magnetic field.
Magnetic North Pole shifts and changes over time in response to changes in the Earth’s magnetic core. It is not a fixed point.

**Comment by umpfmpf** [comment]
When I navigate I have a charger connected anyway, so the battery power consumption doesn't really matter to me.

**Reply by devemux86** [reply]
@umpfmpf 

Compass true north is for the normal map view.

I'm not sure if we also need it in navigation / follow when map rotation is active.
(could complicate things)

**Reply by umpfmpf** [reply]
For me, the compass is actually very important for my orientation. I just don't understand what is meant by true north.

**Reply by devemux86** [reply]
@umpfmpf I updated the first post with info about true north and magnetic north.

In navigation / follow the compass shows the direction of the rotated map's north.
Not the earth's north, although we could do it, but I don't know if we really need it.

**Reply by umpfmpf** [reply]
Would that make a difference as long as I'm somewhere in Central Europe?

**Reply by devemux86** [reply]
@umpfmpf 

Better try Cruiser navigation (or simulation) or follow modes and see the compass.

Then try any compass app from Google Play, like [GPS Status](https://play.google.com/store/apps/details?id=com.eclipsim.gpsstatus2) or [GPSTest](https://play.google.com/store/apps/details?id=com.android.gpstest) apps.
It will help understand what the compass can display.

**Comment by Ciutnik** [comment]
Here is a clever magnetic declination calculator. Once it has determined your position, it shows the compass with "both norths" and the value of their difference:

https://www.ngdc.noaa.gov/geomag/calculators/mobileDeclination.shtml

**Comment by devemux86** [comment]
Compass shows true north is available in the latest:
- https://github.com/devemux86/cruiser/discussions/610

Activate with "Settings | Map | Compass sensor".
For more instructions please see [above](https://github.com/devemux86/cruiser/discussions/628#discussion-7137096).

**Reply by kds112** [reply]
Thanks a lot! The German translation might be changed to "Kompasssensor aktivieren" instead of "Aktivieren Sie den Kompasssensor".

**Reply by devemux86** [reply]
@kds112 thanks, we can change the translation. :+1:

**Comment by devemux86** [comment, marked as answer]
- Compass shows true north is available in https://github.com/devemux86/cruiser/discussions/641

Activate with "Settings | Map | Compass sensor".
For more instructions please see [above](https://github.com/devemux86/cruiser/discussions/628#discussion-7137096).

**Comment by devemux86** [comment]
- Compass sensor filtering was improved in https://github.com/devemux86/cruiser/discussions/641
